<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2024-01-15 Mon 16:29 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>@@html:&lt;img src="pic.jpeg"&gt;@@ Barış Salman</title>
<meta name="author" content="Barış Salman" />
<meta name="generator" content="Org Mode" />
<style>
  #content { max-width: 60em; margin: auto; }
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #e6e6e6;
    border-radius: 3px;
    background-color: #f2f2f2;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
  }
  pre.src:before {
    display: none;
    position: absolute;
    top: -8px;
    right: 12px;
    padding: 3px;
    color: #555;
    background-color: #f2f2f299;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-authinfo::before { content: 'Authinfo'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { }
</style>

<script async src="https://analytics.umami.is/script.js" data-website-id="c0d0b77c-5c98-4814-9721-bb72437e9467"></script>
<link rel="shortcut icon" href="https://omics.sbs/static/base/img/fav.ico" />
<link rel="stylesheet" href="https://omics.sbs/static/base/css/latex.css" />
<link rel="stylesheet" href="https://omics.sbs/static/base/css/style.css" />
<link rel="stylesheet" type="text/css" href="cv.css" />
<script>
// @license magnet:?xt=urn:btih:1f739d935676111cfff4b4693e3816e664797050&amp;dn=gpl-3.0.txt GPL-v3-or-Later
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
// @license-end
</script>
</head>
<body>
<div id="preamble" class="status">

<script>
if (localStorage.darkMode == "true") {
  document.body.classList.add("latex-dark");
}
else {
  document.body.classList.remove("latex-dark");
}
if (localStorage.typeface == "Libertinus") {
    document.body.classList.add("libertinus")
}
else {
    document.body.classList.remove("libertinus")
}
</script>

    <ul id="navbar">
        <li>
            <a href="https://omics.sbs/">Home</a>
        </li>
        <li>
            <a href="https://omics.sbs/blog">Blog</a>
        </li>
        <li>
            <a href="https://omics.sbs/notes">Notes</a>
        </li>
        <li>
            <a href="https://omics.sbs/bioscripts">Bioscripts</a>
        </li>
        <li class="preferences">
            <button class="dark-mode-button" id="dark-mode-toggle" aria-label="Toggle color mode"
                title="Toggle color mode">
                <div class="sun">
                </div>
                <div class="moon">
                    <div class="star"></div>
                    <div class="star small"></div>
                </div>
            </button>
        </li>
        <li class="preferences">
            <input type="button" id="typeface-toggle" title="" value="Aa">
        </li>
    </ul>
</div>
<div id="content" class="content">
<h1 class="title"><img src="pic.jpeg"> Barış Salman
<br />
<span class="subtitle">Curriculum vitae  <br> <a href="mailto:barslmn@gmail.com">E-mail</a> <a href="tel:+905343878420">Phone</a> <a href="https://omics.sbs">Website</a> <br> <a href="https://omics.sbs/github">Github</a> <a href="https://omics.sbs/linkedin">Linkedin</a> <a href="https://omics.sbs/biostars">Biostars</a> <a href="https://omics.sbs/orcid">Orcid</a></span>
</h1>
<div id="outline-container-org4031739" class="outline-2">
<h2 id="org4031739">Technical skills</h2>
<div class="outline-text-2" id="text-org4031739">
</div>
<div id="outline-container-org379911e" class="outline-3">
<h3 id="org379911e">Wet lab and Bioinformatics</h3>
<div class="outline-text-3" id="text-org379911e">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>DNA/RNA/Protein</b></td>
<td class="org-left">Isolation, quantative/qualitative PCRs, Blotting, Cytogenetics and Karyotyping</td>
<td class="org-left"><b>Sequencing</b></td>
<td class="org-left">WGS, WES, Panels, RNAseq, scRNAseq, Sanger</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Functional Studies</b></td>
<td class="org-left">Cell Culture, <i>in vivo</i> models, Microscopy</td>
<td class="org-left"><b>Micro-arrays</b></td>
<td class="org-left">Genotyping, Expression</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Genetic Engineering</b></td>
<td class="org-left">Recombinant DNA technologies</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-orgd5f930f" class="outline-3">
<h3 id="orgd5f930f">Computers and Data Analyses</h3>
<div class="outline-text-3" id="text-orgd5f930f">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Development</b></td>
<td class="org-left">Python, R, Unix Shell</td>
<td class="org-left"><b>Analysis</b></td>
<td class="org-left">Python Scientific Toolkit(numpy, pandas, scipy etc.), R language, Shell Scripting, Bioinformatic tools</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">Machine Learning, PCA, Clustering</td>
</tr>

<tr>
<td class="org-left"><b>Workflow</b></td>
<td class="org-left">Linux, Doom Emacs, Git, Jupyter</td>
<td class="org-left"><b>Statistics and Probability</b></td>
<td class="org-left">Hypothesis testing, Bayesian Theorem, Data distributions</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Writing</b></td>
<td class="org-left">Orgmode, Zotero, Bibtex, LaTeX</td>
<td class="org-left"><b>Visualization</b></td>
<td class="org-left">Matplotlib, Plotly/Dash, LaTeX</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Web Development</b></td>
<td class="org-left">Django, Javascript</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="outline-container-org4757085" class="outline-2">
<h2 id="org4757085">Education</h2>
<div class="outline-text-2" id="text-org4757085">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Doctorate, 2021—ongoing</b></td>
<td class="org-left">Genetics, Institute of Health Sciences, Istanbul University, Turkey</td>
<td class="org-left"><b>GPA 3.5</b></td>
</tr>

<tr>
<td class="org-left"><b>Thesis</b></td>
<td class="org-left">Investigating the Genetic Basis of Epilepsies with Genomic Approaches</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Adviser</b></td>
<td class="org-left">Prof. Dr. Sibel Uğur İşeri</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Notable Courses</b></td>
<td class="org-left">Advanced Molecular Genetics, Mendelian and non Mendelian Diseases, Chromosomal Diseases, Systems Biology</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Master, 2016—2019</b></td>
<td class="org-left">Genetics, Institute of Health Sciences, Istanbul University, Turkey</td>
<td class="org-left"><b>GPA 2.14</b></td>
</tr>

<tr>
<td class="org-left"><b>Thesis</b></td>
<td class="org-left">Study of Split Hand/Foot Malformation with Genomic Techniques and Bioinformatic Approaches</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Adviser</b></td>
<td class="org-left">Assoc Prof. Dr. Sibel Uğur İşeri</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Notable Courses</b></td>
<td class="org-left">Bioinformatics, Genetic Counseling, Medical Genetics, Hospital Rotation for Genetic Counseling</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Bachelor, 2011—2016</b></td>
<td class="org-left">Molecular Biology and Genetics, Istanbul University, Turkey</td>
<td class="org-left"><b>GPA 2.44</b></td>
</tr>

<tr>
<td class="org-left"><b>Thesis</b></td>
<td class="org-left">Metabolic Regulation in Prokaryotes with Small Non-Coding RNAs</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Adviser</b></td>
<td class="org-left">Asst. Prof. Dr. Semian Karaer Uzuner</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Notable Courses</b></td>
<td class="org-left">Methods Used in Molecular Biology(lab), Molecular Genetics, Genetic Engineering</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-org2745909" class="outline-2">
<h2 id="org2745909">Jobs</h2>
<div class="outline-text-2" id="text-org2745909">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Company</b></td>
<td class="org-left">Refgen Biotechnology</td>
<td class="org-left">Development of platform for bioinformatics</td>
</tr>

<tr>
<td class="org-left"><b>Date</b></td>
<td class="org-left">Mar 2021—Ongoing</td>
<td class="org-left">Pipeline research and development</td>
</tr>

<tr>
<td class="org-left"><b>Position</b></td>
<td class="org-left">Genetics and Bioinformatics Specialist</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Company</b></td>
<td class="org-left">Gen-Era Diagnostics</td>
<td class="org-left">Development of pipelines for NGS</td>
</tr>

<tr>
<td class="org-left"><b>Date</b></td>
<td class="org-left">Oct. 2018—Mar 2021</td>
<td class="org-left">Variant pathogenity classification automatization</td>
</tr>

<tr>
<td class="org-left"><b>Position</b></td>
<td class="org-left">Genetics and Bioinformatics Specialist</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Company</b></td>
<td class="org-left">Aktif Gen</td>
<td class="org-left">Development of bioinformatic pipelines</td>
</tr>

<tr>
<td class="org-left"><b>Date</b></td>
<td class="org-left">Jan. 2018—Oct 2018</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Position</b></td>
<td class="org-left">R&amp;D Personnel</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>
</div>
<div id="outline-container-orga0984b9" class="outline-3">
<h3 id="orga0984b9">Internships &amp; Rotations</h3>
<div class="outline-text-3" id="text-orga0984b9">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Genetic Counseling</b></td>
<td class="org-left">Fall 2018</td>
</tr>

<tr>
<td class="org-left"><code>rotation</code></td>
<td class="org-left">IU Cerrahpaşa, Department of Pediatric Genetics and Teratology</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Genetic Counseling</b></td>
<td class="org-left">Spring 2018</td>
</tr>

<tr>
<td class="org-left"><code>rotation</code></td>
<td class="org-left">IU Cerrahpaşa, Department of Medical Genetics</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Cell Culture</b></td>
<td class="org-left">August 2016</td>
</tr>

<tr>
<td class="org-left"><code>internship</code></td>
<td class="org-left">IU Cerrahpasa Department of Medical Biology</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Genomics</b></td>
<td class="org-left">June-July 2015</td>
</tr>

<tr>
<td class="org-left"><code>internship</code></td>
<td class="org-left">IU A.S. Institute of Exprmental Medicine Department of Genetics</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Cytogenetics</b></td>
<td class="org-left">June-July 2014</td>
</tr>

<tr>
<td class="org-left"><code>internship</code></td>
<td class="org-left">IU Department of Medical Genetics</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Molecular Dynamics</b></td>
<td class="org-left">2012-2013</td>
</tr>

<tr>
<td class="org-left"><code>internship</code></td>
<td class="org-left">IU Faculty of Science Quantum Technologies Laboratory</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-orgb336232" class="outline-3">
<h3 id="orgb336232">Certificates</h3>
<div class="outline-text-3" id="text-orgb336232">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Use and care of</b></td>
<td class="org-left">Mar 21 - Apr 08</td>
</tr>

<tr>
<td class="org-left"><b>laboratory animals</b></td>
<td class="org-left">IU HADYEK 46. Laboratory Animal Use Training</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Liquid handling</b></td>
<td class="org-left">2-5 November 2021</td>
</tr>

<tr>
<td class="org-left"><b>robot</b></td>
<td class="org-left">Hamilton Venus Online Advanced Software Training</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">9-12 August 2021</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">Hamilton Venus Online Basic Software Training</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="outline-container-orgaa4746c" class="outline-2">
<h2 id="orgaa4746c">Publications</h2>
<div class="outline-text-2" id="text-orgaa4746c">
<ol class="org-ol">
<li>Susgun, S. et al. Reanalysis of exome sequencing data reveals a treatable neurometabolic origin in two previously undiagnosed siblings with neurodevelopmental disorder. Neurol Sci 44, 2527–2540 (2023).</li>
<li>Susgun, S. et al. Targeted resequencing reveals high-level mosaicism for a novel frameshift variant in WDR45 associated with beta-propeller protein-associated neurodegeneration. Int J Neurosci 1–6 (2023) doi:10.1080/00207454.2023.2208279.</li>
<li>Oguz-Akarsu, E., Salman, B., Ugur-Iseri, S. &amp; Baykan, B. An Extraordinary EEG Phenomenon Misdiagnosed as Nonconvulsive Status Epilepticus: Frequent Subclinical Periodic Discharges Terminated by Sudden Auditory Stimuli. Clin EEG Neurosci 15500594221129965 (2022)</li>
</ol>

<p>
doi:10.1177/15500594221129965.
</p>
<ol class="org-ol">
<li>Mercan, S. et al. Clinical and genetic analyses in syndromic intellectual disability with primary microcephaly reveal biallelic and de novo variants in patients with parental consanguinity. Genes Genom (2022) doi:10.1007/s13258-022-01344-8.</li>
<li>Khalilov, D. et al. Epilepsy or neurodevelopmental disorders are associated with homozygous and pathogenic ELP2 variation in three siblings. Neurocase 28, 488–492 (2022).</li>
<li>Uğur İşeri, S. Combined Analysis Of Linkage And Whole Exome Sequencing Reveals CIC As A Candidate Gene For Isolated Dystonia. Journal of Istanbul Faculty of Medicine 84, 457–463 (2021).</li>
<li>Haryanyan, G. et al. The rare rs769301934 variant in NHLRC1 is a common cause of Lafora disease in Turkey. J Hum Genet 66, 1145–1151 (2021).</li>
<li>Bekdik Şirinocak, P. et al. Susceptibility to Juvenile Myoclonic Epilepsy Associated with the EFHC1 Gene: First Case Report in Turkey. tnd 25, 233–236 (2019).</li>
<li>Akçakaya, N. H. et al. A Novel and Mosaic WDR45 Nonsense Variant Causes Beta-Propeller Protein-Associated Neurodegeneration Identified Through Whole Exome Sequencing and X chromosome Heterozygosity Analysis. Neuromolecular Med 21, 54–59 (2019).
Akçakaya, N. H., Salman, B., Görmez, Z., Tarkan Argüden, Y., Çırakoğlu, A., Çakmur, R., Dönmez Çolakoğlu, B., Hacıhanefioğlu, S., Özbek, U., Yapıcı, Z., &amp; Uğur İşeri, S. A. (2019). A Novel and Mosaic WDR45 Nonsense Variant Causes Beta-Propeller Protein-Associated Neurodegeneration Identified Through Whole Exome Sequencing and X chromosome Heterozygosity Analysis. Neuromolecular Medicine, 21(1), 54–59. <a href="https://doi.org/10.1007/s12017-018-08522-6">https://doi.org/10.1007/s12017-018-08522-6</a></li>
</ol>
</div>
<div id="outline-container-org844e3e5" class="outline-3">
<h3 id="org844e3e5">in pipeline</h3>
<div class="outline-text-3" id="text-org844e3e5">
<ul class="org-ul">
<li>Whole exome sequencing widens the spectrum of associated genes in a cohort with lateral temporal lobe epilepsy <b>(in review.)</b></li>
<li>Variant Version Control: A Git Framework For Keeping Track Of Variant Annotation Changes <b>(in prep.)</b></li>
</ul>
</div>
</div>
<div id="outline-container-org5748e36" class="outline-3">
<h3 id="org5748e36">Collaborative</h3>
<div class="outline-text-3" id="text-org5748e36">
<ul class="org-ul">
<li><a href="https://pubmed.ncbi.nlm.nih.gov/?term=epi25+collaborative">Epi25 Collaborative</a></li>
</ul>
</div>
</div>
<div id="outline-container-org3e91c85" class="outline-3">
<h3 id="org3e91c85">Posters</h3>
<div class="outline-text-3" id="text-org3e91c85">
<ol class="org-ol">
<li>Susgun S., Kesim Y., Salman B., Yucesan E., Khalilov D., Sirin G., Baykan B, Bebek N., Iseri Ugur S., <b>Two candidates bi-allelic variant to neurodevelopmental disorder in a consanguineous family from Turkey</b>, ESHG 2022</li>
<li>Acar A., Say M., Salman B., Dulger M.V., <b>Comparison of Established Microsatellite Instability Detection Tools in Next Generation Sequencing</b>, ESHG 2021</li>
<li>Ugur Iseri S., Akçakaya N. H., Salman B., et al., <b>Exome Sequencing Identifies a FBXO38 Variant Inherited from a Mosaic Mother to cause Distal Hereditary Neuronopathy Type IID with Distinct Features</b>, ASHG 2017</li>
</ol>
</div>
</div>
<div id="outline-container-org3eccd00" class="outline-3">
<h3 id="org3eccd00">Congresses</h3>
<div class="outline-text-3" id="text-org3eccd00">
<ol class="org-ol">
<li>Investigation of Gene Variants Associated with Syndromic Intellectual Disability, 7. International Congress of Cerebral Palsy and Developmental Disorders, February 23-25 2018, Istanbul, Turkey</li>
</ol>
</div>
</div>
</div>
<div id="outline-container-org1614f0b" class="outline-2">
<h2 id="org1614f0b">Projects</h2>
<div class="outline-text-2" id="text-org1614f0b">
<ol class="org-ol">
<li>2021-2022 TUBITAK 1501 Project, 3210420, <b>Modeling and Automatic Analysis of Genotype-Phenotype Related Copy Number Variations</b> Researcher</li>
<li>2021-2022 Research Fund of Istanbul University, ONAP-37862, <b>Transcriptomics Approaches To Biomarker Potential Of  B Cell Expression Analyses In Anti-Neuronal Antibody Related Autoimmune Epilepsy</b> Researcher</li>
<li>2018-2019 Research Fund of Istanbul University, TYL-2018-30315, <b>Study of Split Hand/Foot Malformation with Genomic Techniques and Bioinformatics Approaches</b> Researcher</li>
<li>2018–2018 TUBITAK 1512 Project, 3210420 <b>Development of Cloud Based Software for Next Generation Amplicon Sequencing Technologies</b>, Jan 2018- Oct 2018 Research Fellow</li>
<li>2017-2018 Research Fund of Istanbul University, TDP-2017-25510, <b>Genetic Analyses of Progressive Myoclonic Epilepsy</b> Researcher</li>
<li>2016–2018 TUBITAK 1001 Project, <b>Investigation of Epileptogenesis After Febrile Seizures</b> Research Fellow</li>
</ol>
</div>
</div>
<div id="outline-container-org3aa0636" class="outline-2">
<h2 id="org3aa0636">Software</h2>
<div class="outline-text-2" id="text-org3aa0636">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>VVC</b></td>
<td class="org-left">Variant Version Control <b>(in development)</b></td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">Tool for tracking changes in variant annotation using git.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>&chi;&Sigma;&Delta;</b></td>
<td class="org-left">Cross-symbol checker</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">This tool checks Ensembl and NCBI annotation files for different genome versions and shows which gene symbol is used. This way a more appropriate gene set can be used to avoid the false negatives in the variant discovery process.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Dove</b></td>
<td class="org-left">Downstream VCF Evaluation</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">Tool for annotating VCF files, multi genome analysis and filtering variants.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Pigeon</b></td>
<td class="org-left">Pipelining Genomic Operations</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">Tool for pipelining bioinformatics tools written considering NGS applications.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>Picus</b></td>
<td class="org-left">Pointed Interpretation of Clinical Variant Significance</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">Tool for classifying sequence variants according to ACMG/AMP criteria.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-org429036b" class="outline-2">
<h2 id="org429036b">Awards</h2>
<div class="outline-text-2" id="text-org429036b">
<p>
III. Rare Neurological Diseases Symposium and Neurogenetics Course, <b>Second place project award</b>, Istanbul, Turkey, 2017
</p>
</div>
</div>
<div id="outline-container-orga38f5e6" class="outline-2">
<h2 id="orga38f5e6">Congresses&amp;Courses</h2>
<div class="outline-text-2" id="text-orga38f5e6">
<ol class="org-ol">
<li>2023 YTÜ Biyogen, II. Bioinformatic Conference, Nov 10, <b>Bioinformatics in uncovering the genetic basis of human diseases</b>  (Oral Presentation)</li>
<li>2023 İSTisNa, Strategies, Genomic Approaches and Data Analysis for Rare and Undiagnosed Diseases Course, Oct 12-13, <b>Quality metrics in sequencing data</b>  (Oral Presentation)</li>
<li>2022 Turkish Society of Medical Genetics, Nov 9-13, <b>Genome-Level Evaluation of Copy Number Changes, One of the Important Risk Factors for Genetic Epilepsy</b>  (Oral Presentation)</li>
<li>2020 ESHG 2020, June 6-9, Virtual Conference, <b>GenerAVI: Variant Interpreter and Genetic Analysis Summary Generator</b> (Poster Presentation)</li>
<li>2019 7th International Congress of the Molecular Biology Association of Turkey, Sep 27-29, Istanbul, Turkey, <b>PICUS: Pointed Interpretation of Clinical Variant Significance</b> (Poster Presentation)</li>
<li>2019 Bioinformatics Days III, Gazi University Faculty of Medicine, May 25, Ankara, Turkey, <b>NGS Pipelines with Python</b> (Oral Presentation)</li>
<li>2019 Erciyes University Faculty of Medicine Genetics Days, February 21-23, Kayseri, Turkey, <b>Making Sense of Human Genome with Databases and Bioinformatic Tools</b> (Oral Presentation)</li>
<li>2018 IV. Rare Neurological Diseases Symposium and Neurogenetics Course, May 31-July 1, Istanbul, Turkey, <b>Analysis of Neurogenetics Data Interactive Training and Practice</b> (Instructor)</li>
<li>2018 11. National Epilepsy Congress, May 3–6, 2018, Dalaman, Turkey, <b>Meta Analysis of <i>SCN1A</i> Gene Variants of 114 Patients from Epi25 Exome Data</b> (Oral Presentation)</li>
<li>2018 7. International Congress of Cerebral Palsy and Developmental Disorders, February 23-25, Istanbul, Turkey, <b>Investigation of Gene Variants Associated with Syndromic Intellectual Disability</b></li>
<li>2017 53. National Neurology Congress, November 24–30, Antalya, Turkey, <b>Genomic Approaches to Intellectual Disability Case with Epilepsy</b> (Oral Presentation)</li>
<li>2017 III. Rare Neurological Diseases Symposium and Neurogenetics Course, July 1–2, Istanbul, Turkey, <b>Profiling Anti-Epileptic Drug Resistance an Genetic Diagnosis with Epilepsy Panel</b> (Poster Presentation)</li>
</ol>
</div>
</div>
<div id="outline-container-org252053f" class="outline-2">
<h2 id="org252053f">General skills</h2>
<div class="outline-text-2" id="text-org252053f">
</div>
<div id="outline-container-org6b3cd93" class="outline-3">
<h3 id="org6b3cd93">Languages</h3>
<div class="outline-text-3" id="text-org6b3cd93">
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<tbody>
<tr>
<td class="org-left"><b>Turkish</b> Mother Tongue</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>English</b> Advanced</td>
<td class="org-left">Fluent</td>
</tr>

<tr>
<td class="org-left">These are the languages I am still learning</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left"><b>French</b> Beginner</td>
<td class="org-left">Basic words and phrases</td>
</tr>

<tr>
<td class="org-left"><b>Japanese</b> Beginner</td>
<td class="org-left">Basic words and phrases</td>
</tr>

<tr>
<td class="org-left"><b>Latin</b> Beginner</td>
<td class="org-left">Basic words and phrases</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">

<script src="https://omics.sbs/static/base/js/script.js"></script>
</div>
</body>
</html>
