<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2024-06-22 Sat 15:51 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Investigating male X heterozygosity in short read sequencing</title>
<meta name="author" content="Barış Salman" />
<meta name="generator" content="Org Mode" />
<style type="text/css">
  #content { max-width: 60em; margin: auto; }
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #e6e6e6;
    border-radius: 3px;
    background-color: #f2f2f2;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
  }
  pre.src:before {
    display: none;
    position: absolute;
    top: -8px;
    right: 12px;
    padding: 3px;
    color: #555;
    background-color: #f2f2f299;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-authinfo::before { content: 'Authinfo'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { }
</style>

<script async src="https://analytics.umami.is/script.js" data-website-id="c0d0b77c-5c98-4814-9721-bb72437e9467"></script>
<link rel="shortcut icon" href="https://omics.sbs/static/base/img/fav.ico" />
<link rel="stylesheet" href="https://omics.sbs/static/base/css/latex.css" />
<link rel="stylesheet" href="https://omics.sbs/static/base/css/style.css" />
<script data-isso="//comments.omics.sbs/" src="//comments.omics.sbs/js/embed.min.js"></script>
<script>
// @license magnet:?xt=urn:btih:1f739d935676111cfff4b4693e3816e664797050&amp;dn=gpl-3.0.txt GPL-v3-or-Later
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
// @license-end
</script>
</head>
<body>
<div id="preamble" class="status">

<script>
if (localStorage.darkMode == "true") {
  document.body.classList.add("latex-dark");
}
else {
  document.body.classList.remove("latex-dark");
}
if (localStorage.typeface == "Libertinus") {
    document.body.classList.add("libertinus")
}
else {
    document.body.classList.remove("libertinus")
}
</script>

    <ul id="navbar">
        <li>
            <a href="https://omics.sbs/">Home</a>
        </li>
        <li>
            <a href="https://omics.sbs/blog">Blog</a>
        </li>
        <li>
            <a href="https://omics.sbs/notes">Notes</a>
        </li>
        <li>
            <a href="https://omics.sbs/bioscripts">Bioscripts</a>
        </li>
        <li class="preferences">
            <button class="dark-mode-button" id="dark-mode-toggle" aria-label="Toggle color mode"
                title="Toggle color mode">
                <div class="sun">
                </div>
                <div class="moon">
                    <div class="star"></div>
                    <div class="star small"></div>
                </div>
            </button>
        </li>
        <li class="preferences">
            <input type="button" id="typeface-toggle" title="" value="Aa">
        </li>
    </ul>
</div>
<div id="content" class="content">
<h1 class="title">Investigating male X heterozygosity in short read sequencing</h1>
<div class="dateauthor" id="orge50b9a0">
<p>
19 Jan 2024 — Barış Salman
</p>

<p>
Links
</p>
<ul class="org-ul">
<li><a href="https://github.com/barslmn/XhetRel">Source repository</a></li>
<li><a href="https://www.biostars.org/p/9578275/">Biostars post</a></li>
</ul>

</div>

<div id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li><a href="#orgc8f522d">1. Introduction</a>
<ul>
<li><a href="#org0de6874">1.1. Calling variant from X chromosome in males</a></li>
</ul>
</li>
<li><a href="#orgb5b22a4">2. Materials and Methods</a>
<ul>
<li><a href="#org36ea8f6">2.1. VCF files from the 1000genome project</a></li>
<li><a href="#org36f0c7a">2.2. Preamble</a></li>
<li><a href="#orgad00cf0">2.3. Getting the sample metadata</a></li>
<li><a href="#org26a07f8">2.4. Getting the exon boundaries</a></li>
<li><a href="#orge84ad53">2.5. Getting the VCF files</a></li>
<li><a href="#orgdb192a8">2.6. Merging the BCF files</a></li>
<li><a href="#orga73503d">2.7. Get the genotypes</a></li>
<li><a href="#org998425c">2.8. Count the genotypes</a></li>
<li><a href="#orga418a0e">2.9. Create gene bed</a></li>
<li><a href="#org558d2e0">2.10. Get genotype counts sum per gene</a></li>
<li><a href="#orgb404c55">2.11. Workflow</a></li>
</ul>
</li>
<li><a href="#orgf2ca729">3. Results</a></li>
<li><a href="#org36cc1e8">4. Discussion</a></li>
<li><a href="#orgfb0f29a">5. References</a></li>
<li><a href="#org1f5d72d">6. Glossary</a></li>
<li><a href="#org714a5be">7. Acronyms</a></li>
<li><a href="#org5bbd735">Comments</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgc8f522d" class="outline-2">
<h2 id="orgc8f522d"><span class="section-number-2">1.</span> Introduction</h2>
<div class="outline-text-2" id="text-1">
<p>
During exome sequencing analyses, we encounter heterozygous variants on the X chromosomes in the male samples.
This is not an accurate representation of biological status as males carry only one X and are hemizygous for both sex chromosomes.
Except for the <a class="org-gls" href="#gls.2" id="glsr.2.1">pseudoautosomal regions</a>, at the beginning and the end of both chromosomes X and Y span around 3Mb together.
This region shows homology between the chromosomes and males are &ldquo;homozygous&rdquo; for the regions.
This homology makes it possible for chromosomes to pair during cell division and the only place where recombination can occur between X and Y. (NO_ITEM_DATA:raudseppPseudoautosomalRegionSex2012)
However, the variants we are seeing are located outside of the Pseudoautosomal region
 (<a class="org-gls" href="#gls.12" id="glsr.12.1">PAR</a>) regions and are spread across chromosome X.
There have also been other groups who come across this and used it to check for sample sex as a form of quality control.(NO_ITEM_DATA:doExomeSequencingIdentifies2015)
There have been a lot of questions about this around the web with no satisfying answers.
</p>

<p>
My question was where do thee variants are originated?
We will be making an empirical observation by looking at the 1000 genome data.
</p>
</div>

<div id="outline-container-org0de6874" class="outline-3">
<h3 id="org0de6874"><span class="section-number-3">1.1.</span> Calling variant from X chromosome in males</h3>
<div class="outline-text-3" id="text-1-1">
<p>
Variant callers assume a diploid genome while calling variants.
Males are haploid for the X chromosome meaning there is only one chromosome.
Which doesn&rsquo;t fit the worldview of the variant callers.
What makes it even harder is the highly repetitive regions and regions that show a high degree of homology to other loci in the X chromosome.
Haplotypecaller has a parameter to give ploidy, but since the data we are dealing with doesn&rsquo;t represent a haploid chromosome fitting it into a hemizygous genotype calls might be a challenge.
<a href="https://gatk.broadinstitute.org/hc/en-us/community/posts/360069340411-Why-does-sample-ploidy-ploidy-option-not-work-in-HaplotypeCaller-or-GenotypeGVCFs">https://gatk.broadinstitute.org/hc/en-us/community/posts/360069340411-Why-does-sample-ploidy-ploidy-option-not-work-in-HaplotypeCaller-or-GenotypeGVCFs</a>
<a href="https://gatk.broadinstitute.org/hc/en-us/community/posts/4407495732635-joint-call-samples-with-different-ploidy-on-chrX-chrY-males-vs-females">https://gatk.broadinstitute.org/hc/en-us/community/posts/4407495732635-joint-call-samples-with-different-ploidy-on-chrX-chrY-males-vs-females</a>
<a href="https://gatk.broadinstitute.org/hc/en-us/community/posts/360057847692-Calling-variants-on-chromosome-Y">https://gatk.broadinstitute.org/hc/en-us/community/posts/360057847692-Calling-variants-on-chromosome-Y</a>
<a href="https://gatk.broadinstitute.org/hc/en-us/community/posts/360077511272-CombineGVCF-on-chrX-and-chrY-hg38-running-for-ever">https://gatk.broadinstitute.org/hc/en-us/community/posts/360077511272-CombineGVCF-on-chrX-and-chrY-hg38-running-for-ever</a>
</p>
</div>
</div>
</div>

<div id="outline-container-orgb5b22a4" class="outline-2">
<h2 id="orgb5b22a4"><span class="section-number-2">2.</span> Materials and Methods</h2>
<div class="outline-text-2" id="text-2">

<div id="org3a033b3" class="figure">
<p><img src="_20240619_105304screenshot.png" alt="_20240619_105304screenshot.png" />
</p>
</div>

<p>
These are whole genome Variant Call Format
 (<a class="org-gls" href="#gls.10" id="glsr.10.1">VCF</a>) files and I can&rsquo;t download them all, luckily bcftools can query specific regions on them over the line which makes it both space and speed-efficient.
The region we are going to query is the X chromosome without the <a class="org-gls" href="#gls.12" id="glsr.12.2">PAR</a> region.
We are excluding the <a class="org-gls" href="#gls.12" id="glsr.12.3">PAR</a> since it&rsquo;s a diploid region.
We are going to further filtrate the variants to be heterozygous and since this all started from ES we are just going to look at the exonic regions.
To make things simpler we are going to look at SNVs since INDELs can be challenging to genotype.
We also want to have a depth enough to decide the genotype and don&rsquo;t want anything not passing filters.
</p>

<p>
We can then merge those files into one, count the number of heterozygous genotypes we see for each position and look at them on IGV.
Later we can group them by the gene and see which genes have the most heterozygous variants.
</p>

<p>
We are going to implement all this workflow in the nextflow.
</p>
</div>

<div id="outline-container-org36ea8f6" class="outline-3">
<h3 id="org36ea8f6"><span class="section-number-3">2.1.</span> VCF files from the 1000genome project</h3>
<div class="outline-text-3" id="text-2-1">
<p>
For this job we are going to use individual variant call files from the project since multi-sample <a class="org-gls" href="#gls.10" id="glsr.10.2">VCF</a> are phased.
Looking at the individual variant calls there are New York Genome Center
 (<a class="org-gls" href="#gls.11" id="glsr.11.1">NYGC</a>) study from 2019.
There are two versions of these with one directory named raw<sub>calls</sub><sub>old</sub> and another directory named raw<sub>calls</sub><sub>updated</sub>.
</p>

<p>
The updated directory has a note related to this study that says:
</p>
<blockquote>
<p>
Chromosome X: male samples were processed through HaplotypeCaller with “&#x2013;ploidy 1” parameter applied to non-<a class="org-gls" href="#gls.12" id="glsr.12.4">PAR</a> regions, and “&#x2013;ploidy 2” parameter applied to PAR1 and PAR2 regions, whereas female samples were processed using “&#x2013;ploidy 2”  parameter applied to the entire chrX (see table below for the exact coordinates and corresponding ploidy settings).
</p>
</blockquote>

<p>
Since we want to see the genotypes called as a diploid we will be looking at the calls from the old directory.
The other different thing about the old directory is that samples are located under the major population directories.
</p>
</div>
</div>

<div id="outline-container-org36f0c7a" class="outline-3">
<h3 id="org36f0c7a"><span class="section-number-3">2.2.</span> Preamble</h3>
<div class="outline-text-3" id="text-2-2">
<p>
I had these as a variable while writing the workflow but looking at it now they don&rsquo;t need to be variables.
But I am not gonna change them either since it does make what&rsquo;s being done a bit more transparent.
</p>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #af3a03;">nextflow</span>.enable.dsl=<span style="color: #8f3f71; font-weight: bold;">2</span>

<span style="color: #af3a03;">params</span>.source = <span style="color: #79740e;">"RefSeq"</span>
<span style="color: #af3a03;">params</span>.assembly = <span style="color: #79740e;">"GRCh38"</span>
<span style="color: #af3a03;">params</span>.version = <span style="color: #79740e;">"latest"</span>
<span style="color: #af3a03;">params</span>.feature = <span style="color: #79740e;">"exon"</span>
<span style="color: #af3a03;">params</span>.annot_url = <span style="color: #79740e;">"https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.gff.gz"</span>
<span style="color: #af3a03;">params</span>.exons = <span style="color: #79740e;">"</span><span style="color: #076678;">${params.source}</span><span style="color: #79740e;">.</span><span style="color: #076678;">${params.assembly}</span><span style="color: #79740e;">.</span><span style="color: #076678;">${params.version}</span><span style="color: #79740e;">.bed.gz"</span>

<span style="color: #af3a03;">params</span>.samples_url = <span style="color: #79740e;">'https://www.internationalgenome.org/api/beta/sample/_search/igsr_samples.tsv'</span>
<span style="color: #af3a03;">params</span>.samples_data = <span style="color: #79740e;">'json=%7B%22fields%22%3A%5B%22name%22%2C%22sex%22%2C%22populations.code%22%5D%7D'</span>
<span style="color: #af3a03;">params</span>.samples_info = <span style="color: #79740e;">"igsr_samples.tsv"</span>

<span style="color: #af3a03;">params</span>.base_url = <span style="color: #79740e;">'https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20190425_NYGC_GATK/raw_calls_old'</span>
<span style="color: #af3a03;">params</span>.genotypes = <span style="color: #79740e;">"genotypes.tsv.gz"</span>
<span style="color: #af3a03;">params</span>.bedgraph = <span style="color: #79740e;">"genotype_counts.bedGraph"</span>

</pre>
</div>
</div>
</div>


<div id="outline-container-orgad00cf0" class="outline-3">
<h3 id="orgad00cf0"><span class="section-number-3">2.3.</span> Getting the sample metadata</h3>
<div class="outline-text-3" id="text-2-3">
<p>
Before everything else we need to know who is male.
1000 genomes project has a download link to their sample data on their portal page.
<a href="https://www.internationalgenome.org/data-portal/sample">https://www.internationalgenome.org/data-portal/sample</a>
</p>

<p>
This link makes an API call to another link with the given columns and returns the data.
Since we are only looking for the sex of the individuals we only need the fields name and sex but since they&rsquo;re under the population directories we are also going to need the population codes.
</p>

<label for="sn-symbol" class="sidenote-toggle">⊕</label>
<input type="checkbox" id="sn-symbol" class="sidenote-toggle">
<div class="sidenote" id="org4f5cd36">
<p>
The characters with preceding percent sign are encoded versions of characters like {, [, &ldquo;, ], }.
Because only ASCII character are allowed in the URLs.
<a href="https://www.w3schools.com/tags/ref_urlencode.ASP">https://www.w3schools.com/tags/ref_urlencode.ASP</a>
{&rdquo;fields&ldquo;:[&rdquo;name&ldquo;,&rdquo;sex&ldquo;,&rdquo;populations.code&ldquo;]}
</p>

</div>

<div class="org-src-container">
<pre class="src src-nextflow">curl \
<span style="color: #79740e;">'https://www.internationalgenome.org/api/beta/sample/_search/igsr_samples.tsv'</span> \
--data-raw <span style="color: #79740e;">'json=%7B%22fields%22%3A%5B%22name%22%2C%22sex%22%2C%22populations.code%22%5D%7D'</span> \
-o <span style="color: #79740e;">"igsr_samples.tsv"</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-org26a07f8" class="outline-3">
<h3 id="org26a07f8"><span class="section-number-3">2.4.</span> Getting the exon boundaries</h3>
<div class="outline-text-3" id="text-2-4">
<p>
Since we are interested in getting the exons, we need to filter for them, and for that, we need the exon boundaries.
We can get it using the annotation file from the RefSeq.
</p>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">createbedfile</span> {
    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span>(<span style="color: #79740e;">"</span><span style="color: #076678;">${params.exons}</span><span style="color: #79740e;">"</span>)

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    wget -q -O- "!{params.annot_url}" |</span>
<span style="color: #79740e;">    zcat |</span>
<span style="color: #79740e;">    grep  'NC_000023' |</span>
<span style="color: #79740e;">    awk -F"\t" -v feature="!{params.feature}" '</span>
<span style="color: #79740e;">        /^#!/ {print}</span>
<span style="color: #79740e;">        /^##/ {next}</span>
<span style="color: #79740e;">        $3 ~ feature {</span>
<span style="color: #79740e;">        sub(/^NC_[0]+/, "chr");</span>
<span style="color: #79740e;">        sub(/^chr23/, "chrX");</span>
<span style="color: #79740e;">        split($1, chrom, ".");</span>
<span style="color: #79740e;">        split($9, info, "ID=");</span>
<span style="color: #79740e;">        split(info[2], id, ";");</span>
<span style="color: #79740e;">        printf "%s\t%s\t%s\t%s\n", chrom[1], $4, $5, id[1]}' |</span>
<span style="color: #79740e;">    grep "NM" |</span>
<span style="color: #79740e;">    gzip -c &gt; "!{params.exons}" &amp;&amp;</span>
<span style="color: #79740e;">    echo "INFO: BED file created at !{params.exons}" ||</span>
<span style="color: #79740e;">    echo "ERROR: An error occurred while creating BED file at !{params.exons}"</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>

<p>
This is an awk command which cleans up the chromosome names and splits the info column to write the exon id to fourth bed column.
</p>
</div>
</div>

<div id="outline-container-orge84ad53" class="outline-3">
<h3 id="orge84ad53"><span class="section-number-3">2.5.</span> Getting the VCF files</h3>
<div class="outline-text-3" id="text-2-5">
<p>
For each of our samples, we create the URL for the <a class="org-gls" href="#gls.10" id="glsr.10.3">VCF</a> file and we pass it to bcftools where we perform the filtering and write it to a BCF file named after the sample.
</p>
<label for="sn-symbol" class="sidenote-toggle">⊕</label>
<input type="checkbox" id="sn-symbol" class="sidenote-toggle">
<div class="sidenote" id="org34c0273">
<p>
BCFs are more performant than <a class="org-gls" href="#gls.10" id="glsr.10.4">VCFs</a>.
<a href="https://www.biostars.org/p/273673/">https://www.biostars.org/p/273673/</a>
</p>

</div>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">getfiltervcf</span> {
    <span style="color: #af3a03;">memory</span> <span style="color: #79740e;">'1 GB'</span>
    <span style="color: #af3a03;">cpus</span> <span style="color: #8f3f71; font-weight: bold;">1</span>
    <span style="color: #af3a03;">errorStrategy</span> <span style="color: #79740e;">'ignore'</span>
    <span style="color: #9d0006;">input</span>:
    <span style="color: #8f3f71;">tuple</span> <span style="color: #076678;">val</span>(sample), val(sex), val(population_code)
    <span style="color: #8f3f71;">path</span> <span style="color: #076678;">exons</span>

    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span> <span style="color: #79740e;">"</span><span style="color: #076678;">${sample}</span><span style="color: #79740e;">.bcf{,.csi}"</span>

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    url="!{params.base_url}/!{population_code}/Sample_!{sample}/analysis/!{sample}.haplotypeCalls.er.raw.vcf.gz"</span>

<span style="color: #79740e;">    bcftools view "$url" \</span>
<span style="color: #79740e;">        -r chrX:2781479-153925834 \</span>
<span style="color: #79740e;">        -T "!{exons}" \\</span>
<span style="color: #79740e;">        -v snps \</span>
<span style="color: #79740e;">        -f .,PASS \</span>
<span style="color: #79740e;">        -i 'GT=="0/1" &amp;&amp; FORMAT/DP&gt;=20' \</span>
<span style="color: #79740e;">        -Ob -o !{sample}.bcf</span>
<span style="color: #79740e;">    bcftools index !{sample}.bcf</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-orgdb192a8" class="outline-3">
<h3 id="orgdb192a8"><span class="section-number-3">2.6.</span> Merging the BCF files</h3>
<div class="outline-text-3" id="text-2-6">
<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">mergebcf</span> {
    <span style="color: #9d0006;">input</span>:
    <span style="color: #8f3f71;">path</span>(bcfs)

    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span> <span style="color: #79740e;">'merged/merged.bcf'</span>

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    mkdir merged/</span>
<span style="color: #79740e;">    bcftools merge *.bcf -Ob -o merged/merged.bcf &amp;&amp;</span>
<span style="color: #79740e;">    echo "INFO: Merged BCF file created at merged.bcf" ||</span>
<span style="color: #79740e;">    echo "ERROR: An error occurred while merging BCF files"</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-orga73503d" class="outline-3">
<h3 id="orga73503d"><span class="section-number-3">2.7.</span> Get the genotypes</h3>
<div class="outline-text-3" id="text-2-7">
<p>
Bcftools is the workhorse of this pipeline.
</p>
<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">getgenotypes</span> {
    <span style="color: #9d0006;">input</span>:
    <span style="color: #8f3f71;">path</span> <span style="color: #79740e;">'merged.bcf'</span>

    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span>(<span style="color: #af3a03;">params</span>.genotypes)

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    bcftools query \</span>
<span style="color: #79740e;">      merged.bcf \</span>
<span style="color: #79740e;">      -f '%CHROM\t%POS[\t%GT]\n' | \</span>
<span style="color: #79740e;">      gzip -c &gt; !{params.genotypes} &amp;&amp;</span>
<span style="color: #79740e;">      echo "INFO: Genotypes file created at !{params.genotypes}" ||</span>
<span style="color: #79740e;">      echo "ERROR: An error occurred while extracting genotypes"</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-org998425c" class="outline-3">
<h3 id="org998425c"><span class="section-number-3">2.8.</span> Count the genotypes</h3>
<div class="outline-text-3" id="text-2-8">
<p>
To count we are using awk, and printing them out in a bedGraph format.
</p>

<label for="sn-symbol" class="sidenote-toggle">⊕</label>
<input type="checkbox" id="sn-symbol" class="sidenote-toggle">
<div class="sidenote" id="org5a3e541">
<p>
<a href="https://genome.ucsc.edu/goldenPath/help/bedgraph.html">https://genome.ucsc.edu/goldenPath/help/bedgraph.html</a>
</p>

</div>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">createbedgraph</span> {
    <span style="color: #9d0006;">input</span>:
    <span style="color: #8f3f71;">path</span>(<span style="color: #af3a03;">params</span>.genotypes)

    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span>(<span style="color: #af3a03;">params</span>.bedgraph)

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    zcat !{params.genotypes} | awk -F"\t" 'BEGIN {</span>
<span style="color: #79740e;">        print "track type=bedGraph visibility=full color=255,153,0 altColor=0,0,0"</span>
<span style="color: #79740e;">    }</span>
<span style="color: #79740e;">    {</span>
<span style="color: #79740e;">      count = 0</span>
<span style="color: #79740e;">      for (i = 3; i &lt;= NF; i++) {</span>
<span style="color: #79740e;">        if ($i == "0/1") count++</span>
<span style="color: #79740e;">      }</span>
<span style="color: #79740e;">      printf "%s\t%s\t%s\t%s\n" $1, $2, $2, count</span>
<span style="color: #79740e;">    }' &gt; !{params.bedgraph} &amp;&amp;</span>
<span style="color: #79740e;">    echo "INFO: BedGraph file created at !{params.bedgraph}" ||</span>
<span style="color: #79740e;">    echo "ERROR: An error occurred while creating BedGraph file"</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-orga418a0e" class="outline-3">
<h3 id="orga418a0e"><span class="section-number-3">2.9.</span> Create gene bed</h3>
<div class="outline-text-3" id="text-2-9">
<p>
This is pretty much the same as <a href="#org26a07f8">what we did with the exons</a> but with the genes.
</p>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">creategenebed</span> {
    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span> <span style="color: #79740e;">"genes.bed"</span>

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    wget -q -O- "!{params.annot_url}" |</span>
<span style="color: #79740e;">    zcat |</span>
<span style="color: #79740e;">    grep  'NC_000023' |</span>
<span style="color: #79740e;">    awk -F"\t" -v feature="gene" '</span>
<span style="color: #79740e;">        /^#!/ {print}</span>
<span style="color: #79740e;">        /^##/ {next}</span>
<span style="color: #79740e;">        $3 ~ feature {</span>
<span style="color: #79740e;">        sub(/^NC_[0]+/, "chr");</span>
<span style="color: #79740e;">        sub(/^chr23/, "chrX");</span>
<span style="color: #79740e;">        split($1, chrom, ".");</span>
<span style="color: #79740e;">        split($9, info, "ID=");</span>
<span style="color: #79740e;">        split(info[2], id, ";");</span>
<span style="color: #79740e;">        printf "%s\t%s\t%s\t%s\n", chrom[1], $4, $5, id[1]}' |</span>
<span style="color: #79740e;">    gzip -c &gt; "genes.bed.gz" &amp;&amp;</span>
<span style="color: #79740e;">    echo "INFO: BED file created at genes.bed" ||</span>
<span style="color: #79740e;">    echo "ERROR: An error occurred while creating BED file at genes.bed"</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-org558d2e0" class="outline-3">
<h3 id="org558d2e0"><span class="section-number-3">2.10.</span> Get genotype counts sum per gene</h3>
<div class="outline-text-3" id="text-2-10">
<p>
This looks long but there is no benefit splitting into separate tasks.
Two similar tasks are being performed here only difference between the commands is the datamash operations we are performing.
<a href="#coderef-datamash_count" class="coderef" onmouseover="CodeHighlightOn(this, 'coderef-datamash_count');" onmouseout="CodeHighlightOff(this, 'coderef-datamash_count');">One counts the number of positions</a> and <a href="#coderef-datamash_sum" class="coderef" onmouseover="CodeHighlightOn(this, 'coderef-datamash_sum');" onmouseout="CodeHighlightOff(this, 'coderef-datamash_sum');">other sums the actual genetype counts</a>.
</p>

<p>
We perform the same operation for both of them by getting the top 20 and merging back the gene positions.
</p>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">process</span> <span style="color: #b57614;">getgenecounts</span> {

    <span style="color: #9d0006;">input</span>:
    <span style="color: #8f3f71;">path</span>(genotype_counts), path(genes_bed)

    <span style="color: #9d0006;">output</span>:
    <span style="color: #8f3f71;">path</span> <span style="color: #79740e;">"genotype_counts_by_gene.txt"</span>

    <span style="color: #9d0006;">shell</span>:
    <span style="color: #79740e;">'''</span>
<span style="color: #79740e;">    bedtools intersect \</span>
<span style="color: #79740e;">        -a !{genotype_counts} \</span>
<span style="color: #79740e;">        -b !{genes_bed} -wb |</span>
<span style="color: #79740e;">        awk '{printf "%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $8}' |</span>
<span id="coderef-datamash_sum" class="coderef-off"><span style="color: #79740e;">        datamash -s -g 5 sum 4 |</span></span>
<span style="color: #79740e;">        sort -k 2,2n &gt; genotype_counts_per_gene_by_sum.txt</span>

<span style="color: #79740e;">    bedtools intersect \</span>
<span style="color: #79740e;">        -a !{genotype_counts} \</span>
<span style="color: #79740e;">        -b !{genes_bed} -wb |</span>
<span style="color: #79740e;">        awk '{printf "%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $8}' |</span>
<span id="coderef-datamash_count" class="coderef-off"><span style="color: #79740e;">        datamash -s -g 5 count 4 |</span></span>
<span style="color: #79740e;">        sort -k 2,2n &gt; genotype_counts_per_gene_by_count.txt</span>

<span style="color: #79740e;">    tail -n 20 genotype_counts_per_gene_by_sum.txt | sort &gt; top_20_genes_by_sum_sorted_by_name.txt</span>
<span style="color: #79740e;">    zcat genes.bed.gz | awk '{printf "%s\t%s\t%s\t%s\n", $4,$1,$2,$3}' | sort -k 1,1 &gt; genes_sorted_by_name.bed</span>
<span style="color: #79740e;">    join top_20_genes_by_sum_sorted_by_name.txt genes_sorted_by_name.bed | awk '{printf "%s\t%s\t%s\t%s\n", $3, $4, $5, $1}'&gt; top_20_genes_by_sum_with_positions.bed</span>

<span style="color: #79740e;">    tail -n 20 genotype_counts_per_gene_by_count.txt | sort &gt; top_20_genes_by_count_sorted_by_name.txt</span>
<span style="color: #79740e;">    zcat genes.bed.gz | awk '{printf "%s\t%s\t%s\t%s\n", $4,$1,$2,$3}' | sort -k 1,1 &gt; genes_sorted_by_name.bed</span>
<span style="color: #79740e;">    join top_20_genes_by_count_sorted_by_name.txt genes_sorted_by_name.bed | awk '{printf "%s\t%s\t%s\t%s\n", $3, $4, $5, $1}'&gt; top_20_genes_by_count_with_positions.bed</span>
<span style="color: #79740e;">    '''</span>
}
</pre>
</div>
</div>
</div>

<div id="outline-container-orgb404c55" class="outline-3">
<h3 id="orgb404c55"><span class="section-number-3">2.11.</span> Workflow</h3>
<div class="outline-text-3" id="text-2-11">
<p>
Here only notable parts are; that we are splitting the samplesheet and filtering by male and collecting the bcf files before passing to them to <code>mergebcf</code> process.
</p>

<div class="org-src-container">
<pre class="src src-nextflow"><span style="color: #9d0006;">workflow</span> {
    getsamplesheet()
    samplesheet = getsamplesheet.out
    samplesheet
        .splitCsv(header:[<span style="color: #79740e;">"sample"</span>, <span style="color: #79740e;">"sex"</span>, <span style="color: #79740e;">"population_code"</span>],
            sep:<span style="color: #79740e;">"\t"</span>,
            skip:<span style="color: #8f3f71; font-weight: bold;">1</span>
        )
        .filter { row -&gt; row.sex == <span style="color: #79740e;">"male"</span> &amp;&amp; row.population_code }
        | set {samples}
    createbedfile()
    bcfs = getfiltervcf(samples, createbedfile.out)
    mergebcf(bcfs.collect())
    getgenotypes(mergebcf.out)
    createbedgraph(getgenotypes.out)
    creategenebed()
    getgenecounts(
        createbedgraph.out
        creategenebed.out
    )
}
</pre>
</div>
</div>
</div>
</div>

<div id="outline-container-orgf2ca729" class="outline-2">
<h2 id="orgf2ca729"><span class="section-number-2">3.</span> Results</h2>
<div class="outline-text-2" id="text-3">
<p>
We found a total of 71553 heterozygous genotype calls in 1223 male samples.
2 pairs of the genes here are overlapping;  <i>PABPC1P3</i> overlaps <i>RLIM</i> and <i>DDX53</i> overlaps <i>PTCHD1-AS</i>.
</p>

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">
<caption class="t-above"><span class="table-number">Table 1:</span> Sum of the heterozygous genotype counts in given gene.</caption>

<colgroup>
<col  class="org-left" />

<col  class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">symbol</th>
<th scope="col" class="org-right">sum of genotype counts</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">SLC25A5</td>
<td class="org-right">30694</td>
</tr>

<tr>
<td class="org-left">RLIM</td>
<td class="org-right">10839</td>
</tr>

<tr>
<td class="org-left">PABPC1P3</td>
<td class="org-right">10824</td>
</tr>

<tr>
<td class="org-left">RBMX</td>
<td class="org-right">10370</td>
</tr>

<tr>
<td class="org-left">XAGE3</td>
<td class="org-right">3537</td>
</tr>

<tr>
<td class="org-left">MAGEC1</td>
<td class="org-right">2120</td>
</tr>

<tr>
<td class="org-left">ZXDB</td>
<td class="org-right">1135</td>
</tr>

<tr>
<td class="org-left">CT45A3</td>
<td class="org-right">937</td>
</tr>

<tr>
<td class="org-left">GAGE2A</td>
<td class="org-right">606</td>
</tr>

<tr>
<td class="org-left">PTCHD1-AS</td>
<td class="org-right">597</td>
</tr>

<tr>
<td class="org-left">DDX53</td>
<td class="org-right">596</td>
</tr>

<tr>
<td class="org-left">SPANXB1</td>
<td class="org-right">572</td>
</tr>

<tr>
<td class="org-left">GAGE12B</td>
<td class="org-right">564</td>
</tr>

<tr>
<td class="org-left">PWWP4</td>
<td class="org-right">517</td>
</tr>

<tr>
<td class="org-left">SPANXD</td>
<td class="org-right">493</td>
</tr>

<tr>
<td class="org-left">VCX3B</td>
<td class="org-right">436</td>
</tr>

<tr>
<td class="org-left">GAGE13</td>
<td class="org-right">393</td>
</tr>

<tr>
<td class="org-left">VCX</td>
<td class="org-right">338</td>
</tr>

<tr>
<td class="org-left">GAGE10</td>
<td class="org-right">310</td>
</tr>

<tr>
<td class="org-left">LRCH2</td>
<td class="org-right">290</td>
</tr>
</tbody>
</table>

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">
<caption class="t-above"><span class="table-number">Table 2:</span> Count of heterozygous positions in given gene.</caption>

<colgroup>
<col  class="org-left" />

<col  class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">symbol</th>
<th scope="col" class="org-right">number of het positions</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">LRCH2</td>
<td class="org-right">169</td>
</tr>

<tr>
<td class="org-left">RBMXL3</td>
<td class="org-right">162</td>
</tr>

<tr>
<td class="org-left">PWWP4</td>
<td class="org-right">120</td>
</tr>

<tr>
<td class="org-left">SLC25A5</td>
<td class="org-right">79</td>
</tr>

<tr>
<td class="org-left">MAGEC1</td>
<td class="org-right">72</td>
</tr>

<tr>
<td class="org-left">ARMCX4</td>
<td class="org-right">63</td>
</tr>

<tr>
<td class="org-left">VCX3A</td>
<td class="org-right">63</td>
</tr>

<tr>
<td class="org-left">MAGEE1</td>
<td class="org-right">58</td>
</tr>

<tr>
<td class="org-left">SPANXB1</td>
<td class="org-right">56</td>
</tr>

<tr>
<td class="org-left">FAM47C</td>
<td class="org-right">48</td>
</tr>

<tr>
<td class="org-left">RLIM</td>
<td class="org-right">47</td>
</tr>

<tr>
<td class="org-left">PABPC1P3</td>
<td class="org-right">41</td>
</tr>

<tr>
<td class="org-left">NKAP</td>
<td class="org-right">40</td>
</tr>

<tr>
<td class="org-left">TEX13C</td>
<td class="org-right">40</td>
</tr>

<tr>
<td class="org-left">RBMX</td>
<td class="org-right">37</td>
</tr>

<tr>
<td class="org-left">TRAPPC2</td>
<td class="org-right">35</td>
</tr>

<tr>
<td class="org-left">FGF13</td>
<td class="org-right">34</td>
</tr>

<tr>
<td class="org-left">GAGE2A</td>
<td class="org-right">34</td>
</tr>

<tr>
<td class="org-left">NR0B1</td>
<td class="org-right">33</td>
</tr>

<tr>
<td class="org-left">VCX</td>
<td class="org-right">33</td>
</tr>
</tbody>
</table>


<div id="orgd3eaf66" class="figure">
<p><img src="_20240619_182836screenshot.png" alt="_20240619_182836screenshot.png" />
</p>
<p><span class="figure-number">Figure 1: </span>Sum of genotype counts with close up to some genes. There are four tracks here. The first track shows the genotype counts along the whole chromosome X. Second track is the major regions in chromosome X (<a class="org-gls" href="#gls.12" id="glsr.12.5">PAR</a>, X-added region
 (<a class="org-gls" href="#gls.14" id="glsr.14.1">XAR</a>), X-conserved region
 (<a class="org-gls" href="#gls.13" id="glsr.13.1">XCR</a>), X-transposed region
 (<a class="org-gls" href="#gls.17" id="glsr.17.1">XTR</a>)). Below that is the low complexity and <a class="org-gls" href="#gls.9" id="glsr.9.1">ampliconic regions</a>. At the bottom top 20 genes with the highest heterozygous genotype counts sum. Close-up IGV screenshots of some genes and regions are also included to give a better sense.</p>
</div>

<p>
To further investigate I run the BLAT tool on IGV on some of the hits. The tracks with the gray bars show the BLAT matches.
</p>

<div id="orga1d425f" class="figure">
<p><img src="IGVscreenshots/SLC25A5.png" alt="SLC25A5.png" />
</p>
</div>

<div id="orgab96f04" class="figure">
<p><img src="IGVscreenshots/SLC25A5_chr2.png" alt="SLC25A5_chr2.png" />
</p>
</div>

<div id="orgeb96773" class="figure">
<p><img src="IGVscreenshots/SLC25A5_chr7.png" alt="SLC25A5_chr7.png" />
</p>
</div>

<div id="org763c596" class="figure">
<p><img src="IGVscreenshots/SLC25A5_chr9.png" alt="SLC25A5_chr9.png" />
</p>
</div>

<div id="org71d4486" class="figure">
<p><img src="IGVscreenshots/RBMX.png" alt="RBMX.png" />
</p>
</div>

<div id="org3b90db8" class="figure">
<p><img src="IGVscreenshots/RLIM.png" alt="RLIM.png" />
</p>
</div>


<div id="org1863ca1" class="figure">
<p><img src="IGVscreenshots/RBMXL3.png" alt="RBMXL3.png" />
</p>
</div>

<div id="orgfc1ec7b" class="figure">
<p><img src="IGVscreenshots/RBMXL3_chr11.png" alt="RBMXL3_chr11.png" />
</p>
</div>
</div>
</div>

<div id="outline-container-org36cc1e8" class="outline-2">
<h2 id="org36cc1e8"><span class="section-number-2">4.</span> Discussion</h2>
<div class="outline-text-2" id="text-4">
<p>
The BLAT results shows exons of <i>SLC25A5</i> in other chromosomes namely 2, 7, and 9 without its introns.
It looks like there have been multiple retrotranspositions events.
The one on the chr7 doesn&rsquo;t have an annotation on RefSeq.
</p>

<p>
Another one with high amount of heterozgosity is the <i>RBMX</i> gene.
There is a naming conviction of gametolog gene symbols: if it ends with &ldquo;X&rdquo;, that gene might have a gametolog on chromosome Y which its symbol ends with &ldquo;Y&rdquo;.
e.g. <i>AMELX</i> and <i>AMELY</i> or <i>PRKX</i> and <i>PRKY</i>.
In this case however, there is no gene named <i>RBMY</i> but multiple <i>RBMY1A</i>, <i>RBMY1B</i> etc.
<i>RBMX</i> is a member of a gene family, which their members reside in chromose Y.
BLAT of our <i>RMBX</i> exon 9 doesn&rsquo;t return any hits to any of the Y chromosome relatives.
But, the <i>RBMX</i> has multiple pseudogenes on chromosomes 1, 4, 6, 9, 11 according to NCBI gene summary like <i>SLC25A5</i> also derived from retrotransposition events.
</p>

<p>
What we are seeing here is the one of the limitations of short read sequencing.
Our genome has the same sequences scatter around with events like retrotransposition, duplications, and translocations.
We can&rsquo;t separate where these reads are coming from and seeing them mistakenly aligned.
</p>

<p>
Here we just looked at the chromosome X but its most likely the same errors happen all around the genome.
There will be other regions with false mappings and sequence errors in the genome that we will inadvertently see as heterozygous.
We are just not looking because they&rsquo;re diploid regions and seeing heterozgous variants is expected.
</p>

<label for="sn-symbol" class="sidenote-toggle">⊕</label>
<input type="checkbox" id="sn-symbol" class="sidenote-toggle">
<div class="sidenote" id="org01ab492">

<div id="org9671f09" class="figure">
<p><img src="_20240620_012744screenshot.png" alt="_20240620_012744screenshot.png" />
</p>
<p><span class="figure-number">Figure 2: </span><a href="https://en.wikipedia.org/wiki/Thomas_Hunt_Morgan">Discovery of gene-chromosome relation</a></p>
</div>

</div>
<p>
Studying sex chromosomes shows us a different view  because of their unique inheritance and biology and makes it possible to further question our models.
</p>
</div>
</div>

<div id="outline-container-orgfb0f29a" class="outline-2">
<h2 id="orgfb0f29a"><span class="section-number-2">5.</span> References</h2>
<div class="outline-text-2" id="text-5">
<style>.csl-entry{text-indent: -1.5em; margin-left: 1.5em;}</style><div class="csl-bib-body">
  <div class="csl-entry">NO_ITEM_DATA:raudseppPseudoautosomalRegionSex2012</div>
  <div class="csl-entry">NO_ITEM_DATA:doExomeSequencingIdentifies2015</div>
</div>
</div>
</div>

<div id="outline-container-org1f5d72d" class="outline-2">
<h2 id="org1f5d72d"><span class="section-number-2">6.</span> Glossary</h2>
<div class="outline-text-2" id="text-6">
<p>
<b><span class="org-glsdef" id="gls.9">Ampliconic region</span></b>&emsp;Large regions (about 30% of the MSY euchromatin) where sequence pairs show greater than 99.9% identity, which is maintained by frequent gene conversion (non-reciprocal transfer)&ensp;<a class="org-glsdef" href="#glsr.9.1">1</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.2">Pseudoautosomal region</span></b>&emsp;Region shared between the allosomes. PARs are located at the end of the chromosomes.&ensp;<a class="org-glsdef" href="#glsr.2.1">1</a>
</p>
</div>
</div>

<div id="outline-container-org714a5be" class="outline-2">
<h2 id="org714a5be"><span class="section-number-2">7.</span> Acronyms</h2>
<div class="outline-text-2" id="text-7">
<p>
<b><span class="org-glsdef" id="gls.11">NYGC</span></b>&emsp;New York Genome Center&ensp;<a class="org-glsdef" href="#glsr.11.1">1</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.12">PAR</span></b>&emsp;Pseudoautosomal region&ensp;<a class="org-glsdef" href="#glsr.12.1">1</a>, <a class="org-glsdef" href="#glsr.12.2">2</a>, <a class="org-glsdef" href="#glsr.12.3">3</a>, <a class="org-glsdef" href="#glsr.12.4">4</a>, <a class="org-glsdef" href="#glsr.12.5">5</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.10">VCF</span></b>&emsp;Variant Call Format&ensp;<a class="org-glsdef" href="#glsr.10.1">1</a>, <a class="org-glsdef" href="#glsr.10.2">2</a>, <a class="org-glsdef" href="#glsr.10.3">3</a>, <a class="org-glsdef" href="#glsr.10.4">4</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.17">XTR</span></b>&emsp;X-transposed region&ensp;<a class="org-glsdef" href="#glsr.17.1">1</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.13">XCR</span></b>&emsp;X-conserved region&ensp;<a class="org-glsdef" href="#glsr.13.1">1</a>
</p>

<p>
<b><span class="org-glsdef" id="gls.14">XAR</span></b>&emsp;X-added region&ensp;<a class="org-glsdef" href="#glsr.14.1">1</a>
</p>
</div>
</div>


<div id="outline-container-org5bbd735" class="outline-2">
<h2 id="org5bbd735">Comments</h2>
<div class="outline-text-2" id="text-org5bbd735">
<section id="isso-thread">
<noscript>Javascript needs to be activated to view comments.</noscript>
</section>
</div>
</div>
</div>
<div id="postamble" class="status">

<script src="https://omics.sbs/static/base/js/script.js"></script>
</div>
</body>
</html>
